[JENKINS-75465] Delete RunIdMigrator as it has been 10 years since this migration

gbhat618 opened the pull request:
See [JENKINS-75465](https://issues.jenkins.io/browse/JENKINS-75465).

---------------

This PR proposes to remove `RunIdMigrator`, which was required when upgrading the controller from version 1.596 (or earlier) to 1.597 (or later).

In Jenkins 1.597, build directories started using build numbers instead of the previous datetime format (`yyyy-MM-dd_HH-mm-ss`) for directory naming. Since these versions were released long ago, the migration logic is no longer necessary.

* Jenkins 1.596 – Released on 2015-01-04 ([changelog](https://www.jenkins.io/changelog-old/#v1.596))
* Jenkins 1.597 – Released on 2015-01-19 ([changelog](https://www.jenkins.io/changelog-old/#v1.597))

<details><summary>All related PRs of various changes around RunIdMigration</summary>

- Original PR introduces this migrator: https://github.com/jenkinsci/jenkins/pull/1379
- PR introduces UnMigrate functionality: https://github.com/jenkinsci/jenkins/commit/711b3325adf629585d60aa5a2bcac1d49b8324ae#diff-0d6badde279147be675402c9ace9f24434ba7942c2bb83ce79f069d0dc5a5eb2
- PR that creates test for legacyIds missing in CLE and REST API created jobs: https://github.com/jenkinsci/jenkins/pull/8758/files
- PR that fixes legacyIds marker file in CLI and REST API: https://github.com/jenkinsci/jenkins/pull/9303
- PR that deletes the unmigrate functionality:  https://github.com/jenkinsci/jenkins/pull/9092

</details>

-----
<details><summary>how did the migration work</summary>

RunIdMigrator handled migration of old build directories to the new format by:
* Deleted the build number symlink pointing to datetime format (`yyyy-MM-dd_HH-mm-ss`) directory. Renaming build directories from datetime format to build numbers.
* Updating `build.xml` by changing the `<id>` field from datetime format to numerical ID and storing the original datetime in the `<timestamp>` field. Removed the `<number>` field as it is not longer required the directory itself is the build number.
* Recording migrated builds in a `legacyIds` marker file.
* Providing a `delete` method to remove entries from `legacyIds` when a build is deleted.
</details>

<details><summary>legacyIds marker file</summary>

* On Job Load: First time after upgrade to 1.597 or higher, all builds with the old folder format were migrated, and their IDs were recorded in `legacyIds` (for supported downgrading and undo migration).
* On New Job Creation: An empty legacyIds file was created. This applied to any job created on Jenkins 1.597 or later, including current versions.
* For Jobs with `legacyIds` Present: These jobs were considered migrated, and no further migration was applied.
</details>

**Plan for `legacyIds` Marker File in Existing Jobs**
* Newly created jobs will no longer contain the `legacyIds` marker file.
* `legacyIds` marker file will be deleted upon Jenkins upgrade to reduce inode usage.

This results in a changelog entry and upgrade guideline warning about downgrading may cause potential slowness in Jenkins startup.

### Testing done

**Verify code usages**

Find usages of legacyIds: [search jenkinsci org](https://github.com/search?q=%28org%3Ajenkinsci%29+%22legacyIds%22&type=code) found in 4 places
* Jenkins core
* Plugins
  * support-core-plugin ([code ref](https://github.com/jenkinsci/support-core-plugin/blob/6b9374b72fcfc313cad2cdce03070a423f7da532/src/main/java/com/cloudbees/jenkins/support/impl/ItemsContent.java#L91-L107)) - fixed in https://github.com/jenkinsci/support-core-plugin/pull/639
  * testcomplete-plugin ([code ref](https://github.com/jenkinsci/testcomplete-plugin/blob/aeedac0cdb3544a3a098b1e9d6f79dc4702810fe/src/main/java/com/smartbear/jenkins/plugins/testcomplete/TcDynamicReportAction.java#L175-L211)) - filed https://github.com/jenkinsci/testcomplete-plugin/pull/21 - however plugin doesn't seem to be kept up to date. We will proceed here without waiting for testcomplete-plugin.
  * ~~unique-id-plugin~~: the `LegacyIdStore` mentioned in this plugin is for it's own migration, and not to do with the `legacyIds` file mentioned here. ([code ref](https://github.com/jenkinsci/unique-id-plugin/blob/a25f74db66ab07a04fda1a6f70faedfe78298c78/src/main/java/org/jenkinsci/plugins/uniqueid/impl/IdStoreMigratorV1ToV2.java#L82-L92))

--------------------

#### Manual testing

<details><summary>Test job execution works after upgrade and later downgrade</summary>

* Create a hello-world job in jenkins 2.502 run two builds.
* Upgrade to a version after RunIdMigrator removal 2.503-snapshot, trigger new builds, and verify functionality.
* Create a new job in 2.503-snapshot, run two builds.
* Downgrade to 2.502, trigger builds, and verify functionality.
</details>

<details><summary>Marker file creation no longer happens for now jobs</summary>

```
gbhat@gbhat-MacBook-Pro jobs ±|test ✗|→ ls -l
total 0
drwxr-xr-x@ 5 gbhat  gbhat  160 Mar 24 16:52 fjob_2_502
drwxr-xr-x@ 5 gbhat  gbhat  160 Mar 24 16:52 fjob_2_503
drwxr-xr-x@ 5 gbhat  gbhat  160 Mar 24 16:52 job_2_502
drwxr-xr-x@ 5 gbhat  gbhat  160 Mar 24 16:52 job_2_503
gbhat@gbhat-MacBook-Pro jobs ±|test ✗|→ find . -type f -name "legacyIds"
gbhat@gbhat-MacBook-Pro jobs ±|test ✗|→
```
</details>


<details><summary>Old jobs' marker file is deleted in first startup upon upgrade</summary>

```
2025-03-25 05:33:41.971+0000 [id=41]	INFO	h.p.b.g.GlobalTimeOutConfiguration#load: global timeout not set
2025-03-25 05:33:42.285+0000 [id=40]	INFO	jenkins.InitReactorRunner$1#onAttained: System config loaded
2025-03-25 05:33:42.285+0000 [id=40]	INFO	jenkins.InitReactorRunner$1#onAttained: System config adapted
2025-03-25 05:33:42.291+0000 [id=39]	INFO	hudson.model.Job#onLoad: Deleting legacyIds file in /.../jhome/jobs/fjob_2_502/builds. See https://issues.jenkins.io/browse/JENKINS-75465 for more information.
2025-03-25 05:33:42.291+0000 [id=44]	INFO	hudson.model.Job#onLoad: Deleting legacyIds file in /.../jhome/jobs/job_2_502/builds. See https://issues.jenkins.io/browse/JENKINS-75465 for more information.
2025-03-25 05:33:42.310+0000 [id=45]	INFO	jenkins.InitReactorRunner$1#onAttained: Loaded all jobs
2025-03-25 05:33:42.313+0000 [id=54]	INFO	jenkins.InitReactorRunner$1#onAttained: Configuration for all jobs updated
2025-03-25 05:33:42.324+0000 [id=33]	INFO	jenkins.InitReactorRunner$1#onAttained: Completed initialization
2025-03-25 05:33:42.342+0000 [id=26]	INFO	hudson.lifecycle.Lifecycle#onReady: Jenkins is fully up and running

```
</details>

<details><summary>Marker file gets created for all jobs when downgraded - from 2.503-snapshot to 2.502</summary>

The file is getting created in https://github.com/jenkinsci/jenkins/blob/ba1219d84d7efc101be84376d23d33e639468cc4/core/src/main/java/jenkins/model/RunIdMigrator.java#L149-L149

```
2025-03-25 06:33:48.253+0000 [id=34]	INFO	jenkins.InitReactorRunner$1#onAttained: System config loaded
2025-03-25 06:33:48.253+0000 [id=34]	INFO	jenkins.InitReactorRunner$1#onAttained: System config adapted
2025-03-25 06:33:48.259+0000 [id=54]	INFO	jenkins.model.RunIdMigrator#migrate: Migrating build records in /.../jhome/jobs/fjob_2_503/builds. See https://www.jenkins.io/redirect/build-record-migration for more information.
2025-03-25 06:33:48.259+0000 [id=36]	INFO	jenkins.model.RunIdMigrator#migrate: Migrating build records in /.../jhome/jobs/fjob_2_502/builds. See https://www.jenkins.io/redirect/build-record-migration for more information.
2025-03-25 06:33:48.259+0000 [id=51]	INFO	jenkins.model.RunIdMigrator#migrate: Migrating build records in /.../jhome/jobs/job_2_503/builds. See https://www.jenkins.io/redirect/build-record-migration for more information.
2025-03-25 06:33:48.259+0000 [id=50]	INFO	jenkins.model.RunIdMigrator#migrate: Migrating build records in /.../jhome/jobs/job_2_502/builds. See https://www.jenkins.io/redirect/build-record-migration for more information.
2025-03-25 06:33:48.284+0000 [id=55]	INFO	jenkins.InitReactorRunner$1#onAttained: Loaded all jobs
2025-03-25 06:33:48.287+0000 [id=42]	INFO	jenkins.InitReactorRunner$1#onAttained: Configuration for all jobs updated
2025-03-25 06:33:48.296+0000 [id=41]	INFO	jenkins.InitReactorRunner$1#onAttained: Completed initialization
2025-03-25 06:33:48.313+0000 [id=26]	INFO	hudson.lifecycle.Lifecycle#onReady: Jenkins is fully up and running
```
</details>

<details><summary>Tested that CloudBees KB article's script works during downgrade, and no migration done if `legacyIds` is precreated</summary>

command execution
```
gbhat@gbhat-MacBook-Pro jobs ±|test ✗|→ #!/bin/bash

if [ ! -d "${JENKINS_HOME}/jobs" ]; then
    echo "JENKINS_HOME=${JENKINS_HOME} seems wrong"
fi

# gfind all Job builds directory with a missing 'legacyIds' file.
gcomm -1 -3 --zero-terminated \
  <(gfind ${JENKINS_HOME}/jobs -depth -type f -iname 'legacyIds' -printf '%h\0' | sort -z) \
  <(gfind ${JENKINS_HOME}/jobs -depth -type d -iname 'builds' -prune -print0 | sort -z) | while read -d $'\0' file;
do
  echo "Migrating build records in ${file}"
  touch "${file}/legacyIds"
done
Migrating build records in /Users/gbhat/CBNotes/general/tasks/t36-runidmigrator-delete/jhome/jobs/fjob_2_502/builds
Migrating build records in /Users/gbhat/CBNotes/general/tasks/t36-runidmigrator-delete/jhome/jobs/fjob_2_503/builds
Migrating build records in /Users/gbhat/CBNotes/general/tasks/t36-runidmigrator-delete/jhome/jobs/job_2_502/builds
Migrating build records in /Users/gbhat/CBNotes/general/tasks/t36-runidmigrator-delete/jhome/jobs/job_2_503/builds
```

jenkins startup logs
```
2025-03-25 06:50:47.456+0000 [id=26]	INFO	jenkins.model.Jenkins#<init>: Starting version 2.502
2025-03-25 06:50:47.484+0000 [id=33]	INFO	jenkins.InitReactorRunner$1#onAttained: Started initialization
2025-03-25 06:50:47.554+0000 [id=35]	INFO	jenkins.InitReactorRunner$1#onAttained: Listed all plugins
2025-03-25 06:50:48.610+0000 [id=51]	INFO	jenkins.InitReactorRunner$1#onAttained: Prepared all plugins
2025-03-25 06:50:48.616+0000 [id=48]	INFO	jenkins.InitReactorRunner$1#onAttained: Started all plugins
2025-03-25 06:50:48.617+0000 [id=54]	INFO	jenkins.InitReactorRunner$1#onAttained: Augmented all extensions
2025-03-25 06:50:48.701+0000 [id=50]	INFO	h.p.b.g.GlobalTimeOutConfiguration#load: global timeout not set
2025-03-25 06:50:49.019+0000 [id=33]	INFO	jenkins.InitReactorRunner$1#onAttained: System config loaded
2025-03-25 06:50:49.019+0000 [id=33]	INFO	jenkins.InitReactorRunner$1#onAttained: System config adapted
2025-03-25 06:50:49.040+0000 [id=54]	INFO	jenkins.InitReactorRunner$1#onAttained: Loaded all jobs
2025-03-25 06:50:49.044+0000 [id=45]	INFO	jenkins.InitReactorRunner$1#onAttained: Configuration for all jobs updated
2025-03-25 06:50:49.053+0000 [id=34]	INFO	jenkins.InitReactorRunner$1#onAttained: Completed initialization
2025-03-25 06:50:49.071+0000 [id=26]	INFO	hudson.lifecycle.Lifecycle#onReady: Jenkins is fully up and running
```

</details>

### Proposed changelog entries

- New jobs will no longer create an empty legacyIds marker file in the builds directory, and existing jobs' legacyIds files will be deleted upon upgrade to reduce inode usage.

### Proposed changelog category

/label removed

### Proposed upgrade guidelines

No action is required. On the first startup after upgrading, you will see an info log: `hudson.model.Job#onLoad: Deleting legacyIds file in /.../JENKINS_HOME/jobs/my-job/builds. See https://issues.jenkins.io/browse/JENKINS-75465 for more information.`. However, if you plan to downgrade after upgrading, you may want to manually create the `legacyIds` file in the builds directory to prevent a potential slow startup. This is because the older version will perform migration checks when the file is missing. For reference, see [this CloudBees KB workaround bash script](https://docs.cloudbees.com/docs/cloudbees-ci-kb/latest/client-and-managed-controllers/jenkins-startup-logs-show-runidmigrator-logs#_workaround).

<!-- Comment:
Leave the proposed upgrade guidelines in the pull request with the "N/A" value if no upgrade guidelines are needed.
The changelog generator relies on the presence of the upgrade guidelines section as part of its data extraction process.
-->

### Submitter checklist

- [x] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [x] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.

### Desired reviewers

@mention

<!-- Comment:
If you need an accelerated review process by the community (e.g., for critical bugs), mention @jenkinsci/core-pr-reviewers.
-->

Before the changes are marked as `ready-for-merge`:

### Maintainer checklist

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [x] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

gbhat618 committed:
Delete RunIdMigrator as it has been 10 years since this migration

jglick commented:
> testcomplete-plugin ([code ref](https://github.com/jenkinsci/testcomplete-plugin/blob/aeedac0cdb3544a3a098b1e9d6f79dc4702810fe/src/main/java/com/smartbear/jenkins/plugins/testcomplete/TcDynamicReportAction.java#L175-L211))

It is difficult to follow what the purpose of this code was (no plugin should ever have needed to know about the format of build directory names since `Run.getRootDir` supplies the same for you), but at any rate so far as I can tell the behavior of this method is identical whether `legacyIds` exists and is empty or if it does not exist: it will return `basePath` (whatever that is).

jglick commented on core/src/main/java/hudson/model/RunMap.java:
```diff
@@ -228,13 +221,7 @@ public R put(R r) {
-            n = runIdMigrator.findNumber(id);
-        }
-        return getByNumber(n);
+        return getByNumber(Integer.parseInt(id));
```
Better to still catch `NumberFormatException` and return `null`, just in case some code calls this method with a nonnumeric argument for some reason. (Would be nice to deprecate this method in favor of something taking an `int`, but that would be a bigger project.)

gbhat618 replied:
create https://issues.jenkins.io/browse/JENKINS-75476 and accordingly noted in the code comment.

jglick commented:
Looks like you missed some files

```console
$ git ls-files|fgrep RunIdMigrator
core/src/main/resources/jenkins/model/RunIdMigrator/UnmigrationInstruction/index.jelly
core/src/main/resources/jenkins/model/RunIdMigrator/UnmigrationInstruction/index_bg.properties
core/src/main/resources/jenkins/model/RunIdMigrator/UnmigrationInstruction/index_de.properties
core/src/main/resources/jenkins/model/RunIdMigrator/UnmigrationInstruction/index_it.properties
core/src/main/resources/jenkins/model/RunIdMigrator/UnmigrationInstruction/index_pt_BR.properties
core/src/main/resources/jenkins/model/RunIdMigrator/UnmigrationInstruction/index_sr.properties
core/src/main/resources/jenkins/model/RunIdMigrator/UnmigrationInstruction/index_sv_SE.properties
core/src/main/resources/jenkins/model/RunIdMigrator/UnmigrationInstruction/index_zh_TW.properties
test/src/test/java/jenkins/model/RunIdMigratorTest.java
```

jglick commented:
(Not sure about that last one; there are two `RunIdMigratorTest.java`s, one unit test in `core` and one `JenkinsRule` in `test`. Maybe Git was getting them mixed up.)

gbhat618 committed:
Remove the unmigrator's jelly files

The unmigrator and associated jelly file was added in https://github.com/jenkinsci/jenkins/commit/711b3325adf629585d60aa5a2bcac1d49b8324ae#diff-0d6badde279147be675402c9ace9f24434ba7942c2bb83ce79f069d0dc5a5eb2
In March 2024, the  unmigrator was deleted via https://github.com/jenkinsci/jenkins/pull/9092; however it appears to have missed to remove the Jelly files. Now deleting the jelly file and other resources.

gbhat618 committed:
Remove the test added for legacyIds REST API and CLIs

This test was written as originally the legacyIds were missing when the job was created from CLI or REST API.
https://github.com/jenkinsci/jenkins/pull/8758/files

Later after the issue was fixed, ensuring CLI and REST API jobs also have legacyIds files, the test was made active,
https://github.com/jenkinsci/jenkins/pull/9303

Now as the entire RunMigrator itself is going away there is no need for this test anymore.

gbhat618 committed:
Still catch the NumberFormatException in getById

As the arg is of type `String` it is not gauranteed if a non number comes somehow.
See https://github.com/jenkinsci/jenkins/pull/10456/files#r2010184507

gbhat618 committed:
Delete the legacyIds file if exists upon jobLoad

See https://github.com/jenkinsci/jenkins/pull/10456#issuecomment-2748112449,
deleting the legacyIds file help reduce the number of inodes.
But when doing the downgrade not having the legacyIds file may impact significantly as mentioned in https://issues.jenkins.io/browse/JENKINS-64356

Currently upon upgrade first time start up will have mesages like,
```
2025-03-25 05:33:41.971+0000 [id=41]	INFO	h.p.b.g.GlobalTimeOutConfiguration#load: global timeout not set
2025-03-25 05:33:42.285+0000 [id=40]	INFO	jenkins.InitReactorRunner$1#onAttained: System config loaded
2025-03-25 05:33:42.285+0000 [id=40]	INFO	jenkins.InitReactorRunner$1#onAttained: System config adapted
2025-03-25 05:33:42.291+0000 [id=39]	INFO	hudson.model.Job#onLoad: Deleting legacyIds file in /.../jhome/jobs/fjob_2_502/builds. See https://issues.jenkins.io/browse/JENKINS-75465 for more information.
2025-03-25 05:33:42.291+0000 [id=44]	INFO	hudson.model.Job#onLoad: Deleting legacyIds file in /.../jhome/jobs/job_2_502/builds. See https://issues.jenkins.io/browse/JENKINS-75465 for more information.
2025-03-25 05:33:42.310+0000 [id=45]	INFO	jenkins.InitReactorRunner$1#onAttained: Loaded all jobs
2025-03-25 05:33:42.313+0000 [id=54]	INFO	jenkins.InitReactorRunner$1#onAttained: Configuration for all jobs updated
2025-03-25 05:33:42.324+0000 [id=33]	INFO	jenkins.InitReactorRunner$1#onAttained: Completed initialization
2025-03-25 05:33:42.342+0000 [id=26]	INFO	hudson.lifecycle.Lifecycle#onReady: Jenkins is fully up and running
```

gbhat618 commented on core/src/main/java/hudson/model/Job.java:
```diff
@@ -206,20 +202,19 @@ public synchronized void save() throws IOException {
+        if (Files.deleteIfExists(buildDirPath.resolve("legacyIds"))) {
+            LOGGER.info("Deleting legacyIds file in " + buildDirPath + ". See https://issues.jenkins"
+                        + ".io/browse/JENKINS-75465 for more information.");
+        }
```
Continuing from [this comment](https://github.com/jenkinsci/jenkins/pull/10456#issuecomment-2748112449):

[JENKINS-64356](https://issues.jenkins.io/browse/JENKINS-64356) discusses slow startup caused by a missing legacyIds file. This appears to happen because runIdMigrator.migrate(buildDir, Jenkins.get().getRootDir()) iterates through all builds to check for necessary migrations.

CloudBees provided a [KB article](https://docs.cloudbees.com/docs/cloudbees-ci-kb/latest/client-and-managed-controllers/jenkins-startup-logs-show-runidmigrator-logs#_workaround) with a workaround to recreate the legacyIds file when it's missing.

Since the `legacyIds` file is deleted here, downgrading to an earlier version could result in slow startup.

gbhat618 replied:
I wrote a note in `changelog` and `upgrade guidelines` to mention about downgrading.

gbhat618 committed:
Remove unneeded import after var usage

gbhat618 commented:
> Looks like you missed some files ...
Also double-check...

Nice thanks for these commands, they will be helpful to me forever 😄 thank you. Currently all usages removed.

Noted down the past PRs (or commits) that introduced code around RunIdMigrator in [the JIRA comment](https://issues.jenkins.io/browse/JENKINS-75465?focusedId=453359&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-453359)

gbhat618 committed:
Merge branch 'master' into JENKINS-75465-remove-run-migrator

gbhat618 commented:
>>testcomplete-plugin

> the behavior of this method is identical whether legacyIds exists and is empty or if it does not exist: it will return basePath (whatever that is)

right, I tried to check and clean it up, filed https://github.com/jenkinsci/testcomplete-plugin/pull/21

gbhat618 commented:
The changelog and upgrade guidelines seem too verbose. Should I shorten them, perhaps by removing details about requiring an intermediate update? Upgrading from such an old version to the latest may not even be possible due to significant changes over the years.

gbhat618 commented:
The two test failures are correct because they are using the builds with previous names. These are zip files, looking into it 👀 

* [hudson.model.JobTest.getArtifactsUpTo](https://ci.jenkins.io/job/Core/job/jenkins/job/PR-10456/4/testReport/hudson.model/JobTest/linux_jdk21___Linux___JDK_21___Build___Test___getArtifactsUpTo/)
   <details><summary>getArtifactsUpTo.zip</summary>

   ```
    Archive:  getArtifactsUpTo.zip
      inflating: config.xml
       creating: jobs/
       creating: jobs/testJob/
       creating: jobs/testJob/builds/
       creating: jobs/testJob/builds/2010-04-28_08-26-00/
       creating: jobs/testJob/builds/2010-04-28_08-26-00/archive/
       creating: jobs/testJob/builds/2010-04-28_08-26-00/archive/foo/
       creating: jobs/testJob/builds/2010-04-28_08-26-00/archive/foo/bar/
     extracting: jobs/testJob/builds/2010-04-28_08-26-00/archive/foo/bar/two.txt
       creating: jobs/testJob/builds/2010-04-28_08-26-00/archive/sub/
       creating: jobs/testJob/builds/2010-04-28_08-26-00/archive/sub/dir/
     extracting: jobs/testJob/builds/2010-04-28_08-26-00/archive/sub/dir/one.txt
     extracting: jobs/testJob/builds/2010-04-28_08-26-00/archive/three.txt
      inflating: jobs/testJob/builds/2010-04-28_08-26-00/build.xml
     extracting: jobs/testJob/builds/2010-04-28_08-26-00/log
      inflating: jobs/testJob/config.xml
   ```
   </details>
* [hudson.tasks.FingerprinterTest.actionSerialization](https://ci.jenkins.io/job/Core/job/jenkins/job/PR-10456/4/testReport/hudson.tasks/FingerprinterTest/linux_jdk21___Linux___JDK_21___Build___Test___actionSerialization/)
   <details><summary>actionSerialization.zip</summary>

   ```
    Archive:  actionSerialization.zip
       creating: jobs/
       creating: jobs/j/
       creating: jobs/j-windows/
       creating: jobs/j-windows/builds/
       creating: jobs/j-windows/builds/2013-06-05_14-30-12/
       creating: jobs/j-windows/builds/2013-06-05_14-30-12/archive/
     extracting: jobs/j-windows/builds/2013-06-05_14-30-12/archive/a
      inflating: jobs/j-windows/builds/2013-06-05_14-30-12/build.xml
     extracting: jobs/j-windows/builds/2013-06-05_14-30-12/changelog.xml
      inflating: jobs/j-windows/builds/2013-06-05_14-30-12/log
      inflating: jobs/j-windows/config.xml
     extracting: jobs/j-windows/nextBuildNumber
       creating: jobs/j/builds/
       creating: jobs/j/builds/2013-06-05_14-30-12/
       creating: jobs/j/builds/2013-06-05_14-30-12/archive/
     extracting: jobs/j/builds/2013-06-05_14-30-12/archive/a
      inflating: jobs/j/builds/2013-06-05_14-30-12/build.xml
     extracting: jobs/j/builds/2013-06-05_14-30-12/changelog.xml
      inflating: jobs/j/builds/2013-06-05_14-30-12/log
      inflating: jobs/j/config.xml
     extracting: jobs/j/nextBuildNumber
   ```
   </details>

Will also look for all other zip files for similar 👀

gbhat618 committed:
Update the zip fixtures with datetime-format build dirs to number-format build dirs

Two archives
---------------
1. actionSerialization.zip

Before
```
→ unzip -l actionSerialization.zip
Archive:  actionSerialization.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
        0  02-11-2017 21:51   jobs/
        0  02-11-2017 21:50   jobs/j/
        0  02-11-2017 21:51   jobs/j-windows/
        0  02-11-2017 21:51   jobs/j-windows/builds/
        0  02-11-2017 21:51   jobs/j-windows/builds/2013-06-05_14-30-12/
        0  02-11-2017 21:51   jobs/j-windows/builds/2013-06-05_14-30-12/archive/
       20  02-11-2017 22:01   jobs/j-windows/builds/2013-06-05_14-30-12/archive/a
      996  02-11-2017 22:05   jobs/j-windows/builds/2013-06-05_14-30-12/build.xml
        6  06-05-2013 11:30   jobs/j-windows/builds/2013-06-05_14-30-12/changelog.xml
      448  06-05-2013 11:30   jobs/j-windows/builds/2013-06-05_14-30-12/log
     1002  02-11-2017 21:51   jobs/j-windows/config.xml
        2  06-05-2013 11:30   jobs/j-windows/nextBuildNumber
        0  02-11-2017 21:50   jobs/j/builds/
        0  06-05-2013 11:30   jobs/j/builds/2013-06-05_14-30-12/
        0  06-05-2013 11:30   jobs/j/builds/2013-06-05_14-30-12/archive/
       12  06-05-2013 11:30   jobs/j/builds/2013-06-05_14-30-12/archive/a
      996  06-05-2013 11:30   jobs/j/builds/2013-06-05_14-30-12/build.xml
        6  06-05-2013 11:30   jobs/j/builds/2013-06-05_14-30-12/changelog.xml
      448  06-05-2013 11:30   jobs/j/builds/2013-06-05_14-30-12/log
      993  06-05-2013 11:30   jobs/j/config.xml
        2  06-05-2013 11:30   jobs/j/nextBuildNumber
---------                     -------
     4931                     21 files
```

After
```
Archive:  actionSerialization.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
        0  03-25-2025 15:49   jobs/
        0  02-11-2017 21:51   jobs/j-windows/
        2  06-05-2013 11:30   jobs/j-windows/nextBuildNumber
        0  03-25-2025 15:48   jobs/j-windows/builds/
        0  03-25-2025 15:48   jobs/j-windows/builds/2/
        0  02-11-2017 21:51   jobs/j-windows/builds/2/archive/
       20  02-11-2017 22:01   jobs/j-windows/builds/2/archive/a
     1015  03-25-2025 15:48   jobs/j-windows/builds/2/build.xml
      448  06-05-2013 11:30   jobs/j-windows/builds/2/log
        6  06-05-2013 11:30   jobs/j-windows/builds/2/changelog.xml
     1002  02-11-2017 21:51   jobs/j-windows/config.xml
        0  02-11-2017 21:50   jobs/j/
        2  06-05-2013 11:30   jobs/j/nextBuildNumber
        0  03-25-2025 15:46   jobs/j/builds/
        0  03-25-2025 15:46   jobs/j/builds/2/
        0  06-05-2013 11:30   jobs/j/builds/2/archive/
       12  06-05-2013 11:30   jobs/j/builds/2/archive/a
     1015  03-25-2025 15:46   jobs/j/builds/2/build.xml
      448  06-05-2013 11:30   jobs/j/builds/2/log
        6  06-05-2013 11:30   jobs/j/builds/2/changelog.xml
      993  06-05-2013 11:30   jobs/j/config.xml
---------                     -------
     4969                     21 files
```

2. getArtifactsUpTo.zip

Before
```
Archive:  getArtifactsUpTo.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
      528  11-11-2009 07:05   config.xml
        0  11-10-2009 06:38   jobs/
        0  04-28-2010 20:55   jobs/testJob/
        0  04-28-2010 20:56   jobs/testJob/builds/
        0  04-28-2010 20:58   jobs/testJob/builds/2010-04-28_08-26-00/
        0  04-28-2010 20:59   jobs/testJob/builds/2010-04-28_08-26-00/archive/
        0  04-28-2010 20:58   jobs/testJob/builds/2010-04-28_08-26-00/archive/foo/
        0  04-28-2010 20:58   jobs/testJob/builds/2010-04-28_08-26-00/archive/foo/bar/
        0  04-28-2010 20:58   jobs/testJob/builds/2010-04-28_08-26-00/archive/foo/bar/two.txt
        0  04-28-2010 20:58   jobs/testJob/builds/2010-04-28_08-26-00/archive/sub/
        0  04-28-2010 20:58   jobs/testJob/builds/2010-04-28_08-26-00/archive/sub/dir/
        0  04-28-2010 20:58   jobs/testJob/builds/2010-04-28_08-26-00/archive/sub/dir/one.txt
        0  04-28-2010 20:59   jobs/testJob/builds/2010-04-28_08-26-00/archive/three.txt
      255  04-28-2010 20:57   jobs/testJob/builds/2010-04-28_08-26-00/build.xml
        0  04-28-2010 20:57   jobs/testJob/builds/2010-04-28_08-26-00/log
      311  04-28-2010 20:55   jobs/testJob/config.xml
---------                     -------
     1094                     16 files
```

After
```
Archive:  getArtifactsUpTo.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
      528  11-11-2009 07:05   config.xml
        0  11-10-2009 06:38   jobs/
        0  04-28-2010 20:55   jobs/testJob/
        0  03-25-2025 15:56   jobs/testJob/builds/
        0  03-25-2025 15:56   jobs/testJob/builds/1/
        0  04-28-2010 20:59   jobs/testJob/builds/1/archive/
        0  04-28-2010 20:59   jobs/testJob/builds/1/archive/three.txt
        0  04-28-2010 20:58   jobs/testJob/builds/1/archive/sub/
        0  04-28-2010 20:58   jobs/testJob/builds/1/archive/sub/dir/
        0  04-28-2010 20:58   jobs/testJob/builds/1/archive/sub/dir/one.txt
        0  04-28-2010 20:58   jobs/testJob/builds/1/archive/foo/
        0  04-28-2010 20:58   jobs/testJob/builds/1/archive/foo/bar/
        0  04-28-2010 20:58   jobs/testJob/builds/1/archive/foo/bar/two.txt
      273  03-25-2025 15:56   jobs/testJob/builds/1/build.xml
        0  04-28-2010 20:57   jobs/testJob/builds/1/log
      311  04-28-2010 20:55   jobs/testJob/config.xml
---------                     -------
     1112                     16 files
```

Git Diffs
----------
1. actionSerialization.zip
```
diff --git a/j-windows/builds/2013-06-05_14-30-12/archive/a b/j-windows/builds/2/archive/a
similarity index 100%
rename from j-windows/builds/2013-06-05_14-30-12/archive/a
rename to j-windows/builds/2/archive/a
diff --git a/j-windows/builds/2013-06-05_14-30-12/build.xml b/j-windows/builds/2/build.xml
similarity index 95%
rename from j-windows/builds/2013-06-05_14-30-12/build.xml
rename to j-windows/builds/2/build.xml
index 0ecfb2f..fc4d2fc 100644
--- a/j-windows/builds/2013-06-05_14-30-12/build.xml
+++ b/j-windows/builds/2/build.xml
@@ -16,7 +16,7 @@
       </record>
     </hudson.tasks.Fingerprinter_-FingerprintAction>
   </actions>
-  <number>2</number>
+  <timestamp>1370422812000</timestamp>
   <startTime>1370457012935</startTime>
   <result>SUCCESS</result>
   <duration>178</duration>
@@ -27,4 +27,4 @@
   <hudsonVersion>1.519-SNAPSHOT (private-06/05/2013 00:16 GMT-jglick)</hudsonVersion>
   <scm class="hudson.scm.NullChangeLogParser"/>
   <culprits class="com.google.common.collect.EmptyImmutableSortedSet"/>
-</build>
\ No newline at end of file
+</build>
diff --git a/j-windows/builds/2013-06-05_14-30-12/changelog.xml b/j-windows/builds/2/changelog.xml
similarity index 100%
rename from j-windows/builds/2013-06-05_14-30-12/changelog.xml
rename to j-windows/builds/2/changelog.xml
diff --git a/j-windows/builds/2013-06-05_14-30-12/log b/j-windows/builds/2/log
similarity index 100%
rename from j-windows/builds/2013-06-05_14-30-12/log
rename to j-windows/builds/2/log
diff --git a/j/builds/2013-06-05_14-30-12/archive/a b/j/builds/2/archive/a
similarity index 100%
rename from j/builds/2013-06-05_14-30-12/archive/a
rename to j/builds/2/archive/a
diff --git a/j/builds/2013-06-05_14-30-12/build.xml b/j/builds/2/build.xml
similarity index 95%
rename from j/builds/2013-06-05_14-30-12/build.xml
rename to j/builds/2/build.xml
index 0a4480b..96bc7f2 100644
--- a/j/builds/2013-06-05_14-30-12/build.xml
+++ b/j/builds/2/build.xml
@@ -16,7 +16,7 @@
       </record>
     </hudson.tasks.Fingerprinter_-FingerprintAction>
   </actions>
-  <number>2</number>
+  <timestamp>1370422812000</timestamp>
   <startTime>1370457012935</startTime>
   <result>SUCCESS</result>
   <duration>178</duration>
@@ -27,4 +27,4 @@
   <hudsonVersion>1.519-SNAPSHOT (private-06/05/2013 00:16 GMT-jglick)</hudsonVersion>
   <scm class="hudson.scm.NullChangeLogParser"/>
   <culprits class="com.google.common.collect.EmptyImmutableSortedSet"/>
-</build>
\ No newline at end of file
+</build>
diff --git a/j/builds/2013-06-05_14-30-12/changelog.xml b/j/builds/2/changelog.xml
similarity index 100%
rename from j/builds/2013-06-05_14-30-12/changelog.xml
rename to j/builds/2/changelog.xml
diff --git a/j/builds/2013-06-05_14-30-12/log b/j/builds/2/log
similarity index 100%
rename from j/builds/2013-06-05_14-30-12/log
rename to j/builds/2/log
```

2. getArtifactsUpTo.zip
```
diff --git a/builds/2010-04-28_08-26-00/archive/foo/bar/two.txt b/builds/1/archive/foo/bar/two.txt
similarity index 100%
rename from builds/2010-04-28_08-26-00/archive/foo/bar/two.txt
rename to builds/1/archive/foo/bar/two.txt
diff --git a/builds/2010-04-28_08-26-00/archive/sub/dir/one.txt b/builds/1/archive/sub/dir/one.txt
similarity index 100%
rename from builds/2010-04-28_08-26-00/archive/sub/dir/one.txt
rename to builds/1/archive/sub/dir/one.txt
diff --git a/builds/2010-04-28_08-26-00/archive/three.txt b/builds/1/archive/three.txt
similarity index 100%
rename from builds/2010-04-28_08-26-00/archive/three.txt
rename to builds/1/archive/three.txt
diff --git a/builds/2010-04-28_08-26-00/build.xml b/builds/1/build.xml
similarity index 85%
rename from builds/2010-04-28_08-26-00/build.xml
rename to builds/1/build.xml
index 8764092..c65adfe 100644
--- a/builds/2010-04-28_08-26-00/build.xml
+++ b/builds/1/build.xml
@@ -1,7 +1,7 @@
 <?xml version='1.0' encoding='UTF-8'?>
 <build>
   <actions/>
-  <number>1</number>
+  <timestamp>1272423360000</timestamp>
   <result>SUCCESS</result>
   <duration>100</duration>
   <keepLog>false</keepLog>
diff --git a/builds/2010-04-28_08-26-00/log b/builds/1/log
similarity index 100%
rename from builds/2010-04-28_08-26-00/log
rename to builds/1/log
```

gbhat618 commented on test/src/test/resources/hudson/tasks/FingerprinterTest/actionSerialization.zip:
Change
* rename the build directory from timestamp to build number
* update the `build.xml`: removed `<number>` field, added `<timestamp>` field.

<details><summary>unzip -l actionSerialization.zip</summary>

Before
```
→ unzip -l actionSerialization.zip
Archive:  actionSerialization.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
        0  02-11-2017 21:51   jobs/
        0  02-11-2017 21:50   jobs/j/
        0  02-11-2017 21:51   jobs/j-windows/
        0  02-11-2017 21:51   jobs/j-windows/builds/
        0  02-11-2017 21:51   jobs/j-windows/builds/2013-06-05_14-30-12/
        0  02-11-2017 21:51   jobs/j-windows/builds/2013-06-05_14-30-12/archive/
       20  02-11-2017 22:01   jobs/j-windows/builds/2013-06-05_14-30-12/archive/a
      996  02-11-2017 22:05   jobs/j-windows/builds/2013-06-05_14-30-12/build.xml
        6  06-05-2013 11:30   jobs/j-windows/builds/2013-06-05_14-30-12/changelog.xml
      448  06-05-2013 11:30   jobs/j-windows/builds/2013-06-05_14-30-12/log
     1002  02-11-2017 21:51   jobs/j-windows/config.xml
        2  06-05-2013 11:30   jobs/j-windows/nextBuildNumber
        0  02-11-2017 21:50   jobs/j/builds/
        0  06-05-2013 11:30   jobs/j/builds/2013-06-05_14-30-12/
        0  06-05-2013 11:30   jobs/j/builds/2013-06-05_14-30-12/archive/
       12  06-05-2013 11:30   jobs/j/builds/2013-06-05_14-30-12/archive/a
      996  06-05-2013 11:30   jobs/j/builds/2013-06-05_14-30-12/build.xml
        6  06-05-2013 11:30   jobs/j/builds/2013-06-05_14-30-12/changelog.xml
      448  06-05-2013 11:30   jobs/j/builds/2013-06-05_14-30-12/log
      993  06-05-2013 11:30   jobs/j/config.xml
        2  06-05-2013 11:30   jobs/j/nextBuildNumber
---------                     -------
     4931                     21 files
```

After
```
Archive:  actionSerialization.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
        0  03-25-2025 15:49   jobs/
        0  02-11-2017 21:51   jobs/j-windows/
        2  06-05-2013 11:30   jobs/j-windows/nextBuildNumber
        0  03-25-2025 15:48   jobs/j-windows/builds/
        0  03-25-2025 15:48   jobs/j-windows/builds/2/
        0  02-11-2017 21:51   jobs/j-windows/builds/2/archive/
       20  02-11-2017 22:01   jobs/j-windows/builds/2/archive/a
     1015  03-25-2025 15:48   jobs/j-windows/builds/2/build.xml
      448  06-05-2013 11:30   jobs/j-windows/builds/2/log
        6  06-05-2013 11:30   jobs/j-windows/builds/2/changelog.xml
     1002  02-11-2017 21:51   jobs/j-windows/config.xml
        0  02-11-2017 21:50   jobs/j/
        2  06-05-2013 11:30   jobs/j/nextBuildNumber
        0  03-25-2025 15:46   jobs/j/builds/
        0  03-25-2025 15:46   jobs/j/builds/2/
        0  06-05-2013 11:30   jobs/j/builds/2/archive/
       12  06-05-2013 11:30   jobs/j/builds/2/archive/a
     1015  03-25-2025 15:46   jobs/j/builds/2/build.xml
      448  06-05-2013 11:30   jobs/j/builds/2/log
        6  06-05-2013 11:30   jobs/j/builds/2/changelog.xml
      993  06-05-2013 11:30   jobs/j/config.xml
---------                     -------
     4969                     21 files
```
</details>

<details><summary>git-diff of the change</summary>

```
diff --git a/j-windows/builds/2013-06-05_14-30-12/archive/a b/j-windows/builds/2/archive/a
similarity index 100%
rename from j-windows/builds/2013-06-05_14-30-12/archive/a
rename to j-windows/builds/2/archive/a
diff --git a/j-windows/builds/2013-06-05_14-30-12/build.xml b/j-windows/builds/2/build.xml
similarity index 95%
rename from j-windows/builds/2013-06-05_14-30-12/build.xml
rename to j-windows/builds/2/build.xml
index 0ecfb2f..fc4d2fc 100644
--- a/j-windows/builds/2013-06-05_14-30-12/build.xml
+++ b/j-windows/builds/2/build.xml
@@ -16,7 +16,7 @@
       </record>
     </hudson.tasks.Fingerprinter_-FingerprintAction>
   </actions>
-  <number>2</number>
+  <timestamp>1370422812000</timestamp>
   <startTime>1370457012935</startTime>
   <result>SUCCESS</result>
   <duration>178</duration>
@@ -27,4 +27,4 @@
   <hudsonVersion>1.519-SNAPSHOT (private-06/05/2013 00:16 GMT-jglick)</hudsonVersion>
   <scm class="hudson.scm.NullChangeLogParser"/>
   <culprits class="com.google.common.collect.EmptyImmutableSortedSet"/>
-</build>
\ No newline at end of file
+</build>
diff --git a/j-windows/builds/2013-06-05_14-30-12/changelog.xml b/j-windows/builds/2/changelog.xml
similarity index 100%
rename from j-windows/builds/2013-06-05_14-30-12/changelog.xml
rename to j-windows/builds/2/changelog.xml
diff --git a/j-windows/builds/2013-06-05_14-30-12/log b/j-windows/builds/2/log
similarity index 100%
rename from j-windows/builds/2013-06-05_14-30-12/log
rename to j-windows/builds/2/log
diff --git a/j/builds/2013-06-05_14-30-12/archive/a b/j/builds/2/archive/a
similarity index 100%
rename from j/builds/2013-06-05_14-30-12/archive/a
rename to j/builds/2/archive/a
diff --git a/j/builds/2013-06-05_14-30-12/build.xml b/j/builds/2/build.xml
similarity index 95%
rename from j/builds/2013-06-05_14-30-12/build.xml
rename to j/builds/2/build.xml
index 0a4480b..96bc7f2 100644
--- a/j/builds/2013-06-05_14-30-12/build.xml
+++ b/j/builds/2/build.xml
@@ -16,7 +16,7 @@
       </record>
     </hudson.tasks.Fingerprinter_-FingerprintAction>
   </actions>
-  <number>2</number>
+  <timestamp>1370422812000</timestamp>
   <startTime>1370457012935</startTime>
   <result>SUCCESS</result>
   <duration>178</duration>
@@ -27,4 +27,4 @@
   <hudsonVersion>1.519-SNAPSHOT (private-06/05/2013 00:16 GMT-jglick)</hudsonVersion>
   <scm class="hudson.scm.NullChangeLogParser"/>
   <culprits class="com.google.common.collect.EmptyImmutableSortedSet"/>
-</build>
\ No newline at end of file
+</build>
diff --git a/j/builds/2013-06-05_14-30-12/changelog.xml b/j/builds/2/changelog.xml
similarity index 100%
rename from j/builds/2013-06-05_14-30-12/changelog.xml
rename to j/builds/2/changelog.xml
diff --git a/j/builds/2013-06-05_14-30-12/log b/j/builds/2/log
similarity index 100%
rename from j/builds/2013-06-05_14-30-12/log
rename to j/builds/2/log
```
</details>

jglick replied:
BTW `@LocalData` supports unpacked directories, which is easier to review changes to going forward.

jglick commented on core/src/main/java/hudson/model/Job.java:
```diff
@@ -206,6 +206,15 @@ public void onLoad(ItemGroup<? extends Item> parent, String name)
             throws IOException {
         super.onLoad(parent, name);
 
+        // see https://github.com/jenkinsci/jenkins/pull/10456#issuecomment-2748112449
```
https://github.com/jenkinsci/jenkins/pull/10456#issuecomment-2748112449 FTR

jglick commented on core/src/main/java/hudson/model/RunMap.java:
```diff
@@ -220,8 +220,13 @@ public R put(R r) {
-        return getByNumber(Integer.parseInt(id));
+        try {
+            return getByNumber(Integer.parseInt(id));
+        } catch (NumberFormatException e) { // see https://issues.jenkins.io/browse/JENKINS-75476
```
https://issues.jenkins.io/browse/JENKINS-75476

gbhat618 committed:
Merge branch 'master' into JENKINS-75465-remove-run-migrator

gbhat618 committed:
Delete the legacyIds file from recover_from_legacy_list.zip

File is: test/src/test/resources/hudson/model/QueueTest/recover_from_legacy_list.zip
Found it by `git ls-files -z | xargs -0 fgrep legacyIds`
Also verified there are no plain files in the repo `find . -type f -name 'legacyIds'`.

Before
```
Archive:  recover_from_legacy_list.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
     1526  02-09-2015 02:53   config.xml
      152  02-09-2015 02:53   hudson.model.UpdateCenter.xml
     1680  02-09-2015 02:53   identity.key.enc
      188  02-09-2015 02:53   jenkins.model.JenkinsLocationConfiguration.xml
        0  02-09-2015 02:53   jobs/
        0  02-09-2015 02:53   jobs/test/
        0  02-09-2015 02:53   jobs/test/builds/
        0  02-09-2015 02:53   jobs/test/builds/legacyIds
      468  02-09-2015 02:53   jobs/test/config.xml
      907  02-09-2015 02:53   nodeMonitors.xml
      611  02-09-2015 17:14   queue.xml
       64  02-09-2015 02:53   secret.key
        0  02-09-2015 02:53   secret.key.not-so-secret
        0  02-09-2015 02:53   secrets/
        0  02-09-2015 02:53   secrets/filepath-filters.d/
     2182  02-09-2015 02:53   secrets/filepath-filters.d/30-default.conf
      272  02-09-2015 02:53   secrets/hudson.util.Secret
       48  02-09-2015 02:53   secrets/jenkins.security.ApiTokenProperty.seed
      256  02-09-2015 02:53   secrets/master.key
      272  02-09-2015 02:53   secrets/org.jenkinsci.main.modules.instance_identity.InstanceIdentity.KEY
        0  02-09-2015 02:53   secrets/whitelisted-callables.d/
      953  02-09-2015 02:53   secrets/whitelisted-callables.d/default.conf
        0  02-09-2015 02:53   userContent/
       84  02-09-2015 02:53   userContent/readme.txt
---------                     -------
     9663                     24 files
```

After
```
Archive:  recover_from_legacy_list.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
     1526  02-09-2015 02:53   config.xml
      152  02-09-2015 02:53   hudson.model.UpdateCenter.xml
     1680  02-09-2015 02:53   identity.key.enc
      188  02-09-2015 02:53   jenkins.model.JenkinsLocationConfiguration.xml
        0  02-09-2015 02:53   jobs/
        0  02-09-2015 02:53   jobs/test/
        0  03-26-2025 14:58   jobs/test/builds/
      468  02-09-2015 02:53   jobs/test/config.xml
      907  02-09-2015 02:53   nodeMonitors.xml
      611  02-09-2015 17:14   queue.xml
       64  02-09-2015 02:53   secret.key
        0  02-09-2015 02:53   secret.key.not-so-secret
        0  02-09-2015 02:53   secrets/
      272  02-09-2015 02:53   secrets/hudson.util.Secret
        0  02-09-2015 02:53   secrets/filepath-filters.d/
     2182  02-09-2015 02:53   secrets/filepath-filters.d/30-default.conf
       48  02-09-2015 02:53   secrets/jenkins.security.ApiTokenProperty.seed
      256  02-09-2015 02:53   secrets/master.key
        0  02-09-2015 02:53   secrets/whitelisted-callables.d/
      953  02-09-2015 02:53   secrets/whitelisted-callables.d/default.conf
      272  02-09-2015 02:53   secrets/org.jenkinsci.main.modules.instance_identity.InstanceIdentity.KEY
        0  02-09-2015 02:53   userContent/
       84  02-09-2015 02:53   userContent/readme.txt
---------                     -------
     9663                     23 files
```

gbhat618 commented on test/src/test/resources/hudson/model/QueueTest/recover_from_legacy_list.zip:
Found the `legacyIds` inside of this after running,
```
git ls-files -z | xargs -0 fgrep legacyIds
find . -type f -name 'legacyIds'
```
fixed now.


<details><summary>unzip -l recover_from_legacy_list.zip</summary>

Before
```
Archive:  recover_from_legacy_list.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
     1526  02-09-2015 02:53   config.xml
      152  02-09-2015 02:53   hudson.model.UpdateCenter.xml
     1680  02-09-2015 02:53   identity.key.enc
      188  02-09-2015 02:53   jenkins.model.JenkinsLocationConfiguration.xml
        0  02-09-2015 02:53   jobs/
        0  02-09-2015 02:53   jobs/test/
        0  02-09-2015 02:53   jobs/test/builds/
        0  02-09-2015 02:53   jobs/test/builds/legacyIds
      468  02-09-2015 02:53   jobs/test/config.xml
      907  02-09-2015 02:53   nodeMonitors.xml
      611  02-09-2015 17:14   queue.xml
       64  02-09-2015 02:53   secret.key
        0  02-09-2015 02:53   secret.key.not-so-secret
        0  02-09-2015 02:53   secrets/
        0  02-09-2015 02:53   secrets/filepath-filters.d/
     2182  02-09-2015 02:53   secrets/filepath-filters.d/30-default.conf
      272  02-09-2015 02:53   secrets/hudson.util.Secret
       48  02-09-2015 02:53   secrets/jenkins.security.ApiTokenProperty.seed
      256  02-09-2015 02:53   secrets/master.key
      272  02-09-2015 02:53   secrets/org.jenkinsci.main.modules.instance_identity.InstanceIdentity.KEY
        0  02-09-2015 02:53   secrets/whitelisted-callables.d/
      953  02-09-2015 02:53   secrets/whitelisted-callables.d/default.conf
        0  02-09-2015 02:53   userContent/
       84  02-09-2015 02:53   userContent/readme.txt
---------                     -------
     9663                     24 files
```

After
```
Archive:  recover_from_legacy_list.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
     1526  02-09-2015 02:53   config.xml
      152  02-09-2015 02:53   hudson.model.UpdateCenter.xml
     1680  02-09-2015 02:53   identity.key.enc
      188  02-09-2015 02:53   jenkins.model.JenkinsLocationConfiguration.xml
        0  02-09-2015 02:53   jobs/
        0  02-09-2015 02:53   jobs/test/
        0  03-26-2025 14:58   jobs/test/builds/
      468  02-09-2015 02:53   jobs/test/config.xml
      907  02-09-2015 02:53   nodeMonitors.xml
      611  02-09-2015 17:14   queue.xml
       64  02-09-2015 02:53   secret.key
        0  02-09-2015 02:53   secret.key.not-so-secret
        0  02-09-2015 02:53   secrets/
      272  02-09-2015 02:53   secrets/hudson.util.Secret
        0  02-09-2015 02:53   secrets/filepath-filters.d/
     2182  02-09-2015 02:53   secrets/filepath-filters.d/30-default.conf
       48  02-09-2015 02:53   secrets/jenkins.security.ApiTokenProperty.seed
      256  02-09-2015 02:53   secrets/master.key
        0  02-09-2015 02:53   secrets/whitelisted-callables.d/
      953  02-09-2015 02:53   secrets/whitelisted-callables.d/default.conf
      272  02-09-2015 02:53   secrets/org.jenkinsci.main.modules.instance_identity.InstanceIdentity.KEY
        0  02-09-2015 02:53   userContent/
       84  02-09-2015 02:53   userContent/readme.txt
---------                     -------
     9663                     23 files
```
</details>

gbhat618 commented:
Failing test `hudson.model.FileParameterValueTest.fileParameter_canStillUse_internalHierarchy` ([ref ci test report](https://ci.jenkins.io/job/Core/job/jenkins/job/PR-10456/7/testReport/hudson.model/FileParameterValueTest/linux_jdk17___Linux___JDK_17___Build___Test___fileParameter_canStillUse_internalHierarchy/))
seems like a flake as only failed in JDK17, same test is also failing in other PR ([ref ci test report](https://ci.jenkins.io/job/Core/job/jenkins/job/PR-10467/2/testReport/hudson.model/FileParameterValueTest/linux_jdk17___Linux___JDK_17___Build___Test___fileParameter_canStillUse_TildeInFileName/))

checking 👀

timja commented:
> The changelog and upgrade guidelines seem too verbose. Should I shorten them, perhaps by removing details about requiring an intermediate update? Upgrading from such an old version to the latest may not even be possible due to significant changes over the years.
> 
> any suggestions about `changelog` and `upgrade guidelines` section?

From my PoV the changelog entry is far too long.

This is fine:

> Removed migration support for upgrading directly from Jenkins 1.596 (2015) or earlier to 1.597+

Upgrade guide can be longer although personally I think only the downgrade instructions are needed after 10 years...

gbhat618 commented:
> From my PoV the changelog entry is far too long ... Upgrade guide can be longer although personally I think only the downgrade instructions are needed after 10 years

Thank you for this. I made the `changelog` smaller, and fixed the `upgrade guidelines` by only keeping a note about downgrade recommendation.

Just checked Jenkins stats (https://stats.jenkins.io/statistics) for versions older than 1.596—about 30 are still in use! 😄 No direct link, so here’s a quick screenshot.
<details><summary>jenkins older to 1.596, stats Oct 2024</summary>

![image](https://github.com/user-attachments/assets/8904f7f2-f2c8-4d14-8023-6da4b3478602)
</details>

gbhat618 commented:
Once https://github.com/jenkinsci/jenkins/pull/10471 is merged, update this branch to trigger the CI build, which is expected to pass.

gbhat618 committed:
Merge branch 'master' into JENKINS-75465-remove-run-migrator

timja reviewed (approved):
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

gbhat618 committed:
Merge branch 'master' into JENKINS-75465-remove-run-migrator
